internal/tracker/seen.go

```diff
@@ -32,169 +33,227 @@ func (k keyKind) String() string {
 // SeenTracker tracks which keys have been seen with which TOML type to flag duplicates
 // and mismatches according to the spec.
 type SeenTracker struct {
-	root    *info
-	current *info
+	entries    []entry
+	currentIdx int
+	nextId     int
```

Comment:
Convention is to use `nextID`

Reply:
TIL, thank you! https://github.com/golang/go/wiki/CodeReviewComments#initialisms